src/store/juju/types.ts

```diff
@@ -25,19 +28,14 @@ export type AdditionalController = {
 
 export type Controllers = Record<string, (Controller | AdditionalController)[]>;
 
-export type ModelInfo =
-  | Omit<JujuModelInfo, "agent-version"> & {
-      "agent-version": string;
-    };
-
-export type ModelData = {
-  applications: FullStatus["applications"];
+// There is some model data that we don't want to store from the full status because it changes
+// too often causing needless re-renders and is currently irrelevant
+// like controllerTimestamp.
+export type ModelData = Omit<
+  FullStatusWithAnnotations,
```

Comment:
We can now use the model data type from jujulib, minus the bits of data we don't want to store.